server/meshmodel/helper.go

```diff
@@ -162,6 +164,7 @@ func (erh *EntityRegistrationHelper) generateRelationships(pathToComponents stri
 // If an error occurs, it logs the error
 func (erh *EntityRegistrationHelper) watchComponents(ctx context.Context) {
 	var err error
+	meshmodel.Mutex.Lock()
```

Comment:
These mutex is for shared variables right used in registryLog function?
If so move this lock smd unlock within that function call.

Reply:
@MUzairS15 
but we are locking as the registering is started when it enters the watchComponent for the first time 
the main aim of this mutex is that the function GetmeshmodelRegistrant shouldn't be displaying the value if the registration is not complete  and unlock it when the registration is complete . That is why we are locking it at the start and then unlocking it when the registration is complete .
```https://github.com/meshery/meshery/blob/e2bbe45b50d5acdb9493bab4fbf90b0998b4db36/server/handlers/component_handler.go#L1378```